specification/mysql/resource-manager/Microsoft.DBforMySQL/2017-04-30-preview/mysql.json

```diff
@@ -1267,6 +1454,74 @@
       },
       "description": "A list of firewall rules."
     },
+    "VirtualNetworkRuleProperties": {
+      "description": "Properties of a virtual network rule.",
+      "required": [
+        "virtualNetworkSubnetId"
+      ],
+      "type": "object",
+      "properties": {
+        "virtualNetworkSubnetId": {
+          "description": "The ARM resource id of the virtual network subnet.",
+          "type": "string"
+        },
+        "ignoreMissingVnetServiceEndpoint": {
+          "description": "Create firewall rule before the virtual network has vnet service endpoint enabled.",
+          "type": "boolean"
+        },
+        "state": {
```

Comment:
Should this be called "provisioningState"? Are these the states which this resource can go into during the provisioning?

Reply:
We are keeping this consistent with SQL APIs , yes a specific VNET rule can be  in one of these states .